### ISSUE TITLE
The medial version of 'DA' is right in the candidate view. But it becomes wrong in editor view or after user copy/paste it to another APP.

### ISSUE THREAD

**Issue by HusilengCode** [suragch/mongol-library#8]
The medial version of 'DA' is right in the candidate's view. But it becomes wrong after users copy/paste it to another APP or view. The Unicode for such a case should be \u1833 \u180B instead of just \u1833.

**Comment by suragch** [comment]
Hi @HusilengCode 

What is the word you are using the medial DA in?

I haven't been developing this library for the last few years since I switched to Flutter for my Android and iOS apps. If you would like to continue the development yourself in your own fork, I can point people your direction.

**Comment by HusilengCode** [comment]
Hi Jonathan, 

ᠰᠨᠳᠤᠨ , actually, it is wrong based on the Mongolian Unicode block, and this bug shows up on most phones.


wrong sadon
![Wrong_sadon](https://user-images.githubusercontent.com/124936308/225877964-de5a83c5-c516-4722-9eb5-508c09d7698b.PNG)

wrong sadon Unicode 
![Wrong_sadon_unicode](https://user-images.githubusercontent.com/124936308/225878227-b5cf71ad-e86a-4fab-83f1-932fae26c937.PNG)

Right sadon
![Right_sadon](https://user-images.githubusercontent.com/124936308/225878089-73fbf50b-de43-4715-a767-b4def9e9b30e.PNG)

Right sadon Unicode 
![Right_sadon_unicode](https://user-images.githubusercontent.com/124936308/225878268-4098ac71-7fb8-4b27-82bb-bf8cd1fd0f40.PNG)

Mongolian Unicode block
![Mongolian unicode block](https://user-images.githubusercontent.com/124936308/225878311-0084fd33-b5ad-4d11-ab8b-0a16fcd967bb.PNG)


Yeah, I can understand. Flutter is compatible with most platforms and saves lots of effort. 
I am currently trying to write an IME because the available options have too many mistakes, and Flutter may not do this job. Users need a true Unicode-based flawless IME.
I tried to go through your code two short times, but it is still hard to follow all logic and details (especially the other one's code).
But do not worry, I will do my homework. You already did a great job! I highly appreciate it. If you need any support from me, please let me know!

Best Regards,
Husileng

**Comment by HusilengCode** [comment]
I labeled the two different medial DA.

![image](https://user-images.githubusercontent.com/124936308/225883118-6c8a0a2a-e264-4516-8d7c-8001505398dd.png)

Best Regards,
Husileng

**Comment by suragch** [comment]
Has this been standardized anywhere? For example, if the image you included above comes from [Wikipedia](https://en.wikipedia.org/wiki/Mongolian_(Unicode_block)), then this is how it looks on my computer:

<img width="405" alt="Screen Shot 2023-03-18 at 15 43 52" src="https://user-images.githubusercontent.com/10300220/226092734-e2313734-86b6-434f-98de-d0b166004cfc.png">

From what I heard, Unicode no longer specifies the variants. (They used to.) Now they just specify the main codes. See the [current specification](https://unicode.org/charts/PDF/U1800.pdf).

On my phone and computer, the "wrong" coding looks correct for ᠰᠨᠳᠤᠨ. Obviously, we are using two different fonts that implement Unicode in different ways. This has been the problem I've had as long as I've been programming for the traditional script. The companies and nations involved still haven't come to an agreement about how to encode the variants.

Anyway, if most users are using the font that you are using, then it would be a worthwhile project to make a new input method. I know what you mean about it being difficult to understand someone else's code so I don't blame you for working on your own project.

**Comment by HusilengCode** [comment]
Ok, as I checked Apple and Google having the same font as the wrong one. Microsoft is fine.

Still for ᠰᠨᠳᠤᠨ. If I put a space after 'DA,' it is ᠰᠨᠳ ᠤᠨ because 'DA' shows in the final version.  And it should go back to the right one ᠰᠨᠳᠤᠨ after I delete the space. It is like this in Microsoft. But our code acts as follows, and the wrong coding comes out at the end:

I did this in Chimee
![image](https://user-images.githubusercontent.com/124936308/226139694-b4811e6c-f63c-402d-8475-ee403a7846eb.png)
![image](https://user-images.githubusercontent.com/124936308/226139714-fb78e9be-7e61-4cb4-8f22-35aac60ce305.png)
![image](https://user-images.githubusercontent.com/124936308/226139730-d62afd8c-7aa3-4b13-a46b-2ce449137c01.png)

**Comment by suragch** [comment]
I just got around to testing this out, but I couldn't reproduce the problem in Chimee. When I enter sadon (\u1830\u1820\u1833\u1824\u1828) and then add a space between \u1833 and \1824 I see the same rendering as you have in your image. But then when I delete the space it automatically returns to the correct rendering (not like your third image. I'm using Chimee 3.3.1.
